src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c

```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+                        this->textId = 0x5F1;
+                    }
+                    func_800B8500(&this->actor, globalCtx, this->actor.xzDistToPlayer, this->actor.yDistToPlayer, 0);
+                    break;
+                case PLAYER_FORM_HUMAN:
+                    this->textId = 0x5F7;
+                    break;
+                case PLAYER_FORM_GORON:
+                case PLAYER_FORM_ZORA:
+                    switch (CURRENT_DAY) {
+                        case 1:
+                            this->textId = 0x5E9;
+                            break;
+                        case 2:
+                            this->textId = 0x5EA;
+                            break;
+                        case 3:
+                            tempDay = gSaveContext.day;
+                            tempTimeBeforeMoonCrash =
+                                ((-(tempDay % 5 << 0x10) - ((u16)(gSaveContext.time - 0x4000))) + 0x40000);
+                            if (tempTimeBeforeMoonCrash < 2730.6667f /* 1 hr */) {
+                                this->textId = 0x5E8;
+                            } else {
+                                this->textId = 0x5EB;
+                            }
+                            break;
+                    }
+            }
+            break;
+    }
+    func_8013E1C8(&this->skelAnime, D_809CDC7C, 2, &this->animationIndex);
+    this->actionFunc = func_809CD328;
+}
+
+void func_809CD328(EnBji01* this, GlobalContext* globalCtx) {
+
+    switch (func_80152498(&globalCtx->msgCtx)) {
+        case 0:
+            Math_ScaledStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0x444);
+            func_809CCDE0(this, globalCtx);
+            if (this->actor.shape.rot.y == this->actor.yawTowardsPlayer) {
+                func_801518B0(globalCtx, this->textId, &this->actor);
+            }
+            break;
+        case 4:
+            if (func_80147624(globalCtx) != 0) {
+                this->actor.flags &= 0xFFFEFFFF;
+                this->actor.params = 1;
+                switch (globalCtx->msgCtx.choiceIndex) {
+                    case 0:
+                        func_8019F208();
+                        func_801477B4(globalCtx);
+                        func_809CD634(this, globalCtx);
+                        break;
+                    case 1:
+                        func_8019F230();
+                        switch (gSaveContext.playerForm) {
+                            case PLAYER_FORM_DEKU:
+                                func_80151938(globalCtx, 0x5F0U);
+                                break;
+                            case PLAYER_FORM_HUMAN:
+                                func_80151938(globalCtx, 0x5F8U);
+                                break;
+                            case PLAYER_FORM_GORON:
+                            case PLAYER_FORM_ZORA:
+                                func_80151938(globalCtx, 0x5E1U);
+                                break;
```

Comment:
There's no need for those extra `U` at the end.
```suggestion
                            case PLAYER_FORM_DEKU:
                                func_80151938(globalCtx, 0x5F0);
                                break;
                            case PLAYER_FORM_HUMAN:
                                func_80151938(globalCtx, 0x5F8);
                                break;
                            case PLAYER_FORM_GORON:
                            case PLAYER_FORM_ZORA:
                                func_80151938(globalCtx, 0x5E1);
                                break;
```